File: 241102/G or H 2/G-or-H-2.py
N = int(input())

arr = [0]*101
for i in range(N):
    p, a = input().split()
    arr[int(p)] = a

M = 0
for i in range(101):
    if arr[i] == 'G' or arr[i] == 'H':
        cntG = 0
        cntH = 0
        if arr[i] == 'G':
            cntG += 1
        if arr[i] == 'H':
            cntH += 1

        d = 0

        for j in range(i+1, 101):
            if arr[j] == 'G':
                cntG += 1
                if cntG == 0 or cntH == 0 or cntG == cntH:
                    d = max(d, j-i)
            elif arr[j] == 'H':
                cntH += 1
                if cntG == 0 or cntH == 0 or cntG == cntH:
                    d = max(d, j-i)
        M = max(M, d)

print(M)

File: 250120/G or H 2/G-or-H-2.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int main() {
    // Please write your code here.
    char arr[101]={}; 
    int S = 100;
    int L = 0;

    int N;
    cin >> N;
    for (int i=0; i<N; i++) {
        int x; char c;
        cin >> x >> c;
        arr[x] = c;

        S = min(S, x);
        L = max(L, x);
    }

    int max_val = 0;
    for (int i=S; i<=L; i++) {
        for (int j=i+1; j<=L; j++) {
            if (!((arr[i] == 'G' || arr[i] == 'H') && (arr[j] == 'G' || arr[j] == 'H'))) continue;

            int cntG = 0;
            int cntH = 0;
            for (int k=i; k<=j; k++) {
                if (arr[k] == 'G') cntG++;
                else if (arr[k] == 'H') cntH++;
            }
            if ((cntG != 0 && cntH == 0) || (cntG == 0 && cntH != 0) || (cntG == cntH)) {
                max_val = max(max_val, j-i);
            }
        }
    }
    cout << max_val;

    return 0;
}